dask_bigquery/core.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from functools import partial
+
+import pandas as pd
+import pyarrow
+from dask.base import tokenize
+from dask.dataframe.core import new_dd_object
+from dask.highlevelgraph import HighLevelGraph
+from dask.layers import DataFrameIOLayer
+from google.cloud import bigquery, bigquery_storage
+
+
+@contextmanager
+def bigquery_client(project_id=None):
+    """This context manager is a temporary solution until there is an
+    upstream solution to handle this.
+    See  googleapis/google-cloud-python#9457
+    and googleapis/gapic-generator-python#575 for reference.
+    """
+    with bigquery.Client(project_id) as bq_client:
```

Comment:
Doesn't have to be this PR, but it would be really helpful if we could attribute these requests to Dask/Dask-BigQuery. https://github.com/coiled/dask-bigquery/issues/6